library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -168,23 +163,39 @@ public DownloadBatch retrieveBatchBy(DownloadInfo downloadInfo) {
             batchesCursor.close();
         }
 
-        if (!forDeletion.isEmpty()) {
-            deleteBatchesForIds(forDeletion, downloads);
+        return batches;
+    }
+
+    public void deleteMarkedBatchesFor(Collection<DownloadInfo> downloads) {
+        Cursor batchesCursor = resolver.query(Downloads.Impl.BATCH_CONTENT_URI, null, SELECT_MARKED_FOR_DELETION, null, null);
```

Comment:
The api contract does not say to use `?` for the right hand side, but if perfmatters: 

Use question mark parameter markers such as 'phone=?' instead of explicit values in the selection parameter, so that queries that differ only by those values will be recognized as the same for caching purposes.

Didn't know about this.
Does it make any change in performance though?